tools/pv_tree_generator/build_tree.py

```diff
@@ -0,0 +1,152 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""build the property-value tree """
+
+import collections
+import util
+import text_format
+
+MAX_LEVEL = 6
+
+def build_tree_recursive(pos, level, pop_obs_spec, stat_vars, show_all, parent=None):
+    """Recursively build the ui tree"""
+    #get the property of the ui node
+    if parent:
+        property_diff = (set(pos.properties) - set(parent.pop_obs_spec.properties)).pop()
+        parent_pv = parent.pv
+    else:
+        property_diff = pos.properties[0]  # This is single pv pos
+        parent_pv = {}
+
+    prop_ui_node = util.UiNode(pos, parent_pv, True, property_diff)
+    result = {
+        'populationType': prop_ui_node.pop_type,
+        'show': 'yes',
+        'selected': 'no',
+        'expanded': 'no',
+        'title': text_format.format_title(prop_ui_node.text),
+        'type': 'Property',
+        'count': 0,
+        'children': [],
+        'sv_set': set()
+    }
+
+    #get the child specs of the current node
+    child_pos = []
+    for c_pos in pop_obs_spec[level+1]:
+        if pos.pop_type == c_pos.pop_type and set(pos.properties) < set(c_pos.properties):
+            child_pos.append(c_pos)
+
+    for sv in stat_vars[pos.key]:
+        if sv.match_ui_node(prop_ui_node):
+            value_ui_pv = collections.OrderedDict()
+            for prop, val in parent_pv.items():
+                value_ui_pv[prop] = val
+            value_ui_pv[property_diff] = sv.pv[property_diff]
+            value_ui_node = util.UiNode(pos, value_ui_pv, False, property_diff)
+
+            value_blob = {
+                'populationType': value_ui_node.pop_type,
+                'show': 'yes',
+                'selected': 'no',
+                'expanded': 'no',
+                'argString': sv.dcid,
+                'title': text_format.format_title(value_ui_node.text),
+                'type': 'value',
+                'enum': value_ui_node.enum,
+                'count': 1,
+                'children': [],
+                'sv_set': set([sv.dcid])
+            }
+            # add statistical variables as the child of current node
+            result['children'].append(value_blob)
+
+            if level <= MAX_LEVEL:
+                #build the branches recursively
+                for child in child_pos:
+                    branch = build_tree_recursive(child, level + 1, pop_obs_spec,
+                                        stat_vars, show_all, value_ui_node)
+                if branch['children']:
+                    value_blob['children'].append(branch)
+                value_blob['sv_set'] |= branch['sv_set']
+                del branch['sv_set']
+            value_blob['count'] = len(value_blob['sv_set'])
+
+    result['children'] = text_format.filter_and_sort(property_diff, result['children'], show_all)
```

Comment:
line too long. Suggest to use some lint tool and enforce line width 80:

http://google.github.io/styleguide/pyguide.html#32-line-length

Reply:
done